tests/unit/classifier/Perceptron_unittest.cc

```diff
@@ -68,16 +72,16 @@ TEST(Perceptron, custom_hyperplane_initialization)
 	auto test_features = wrap(env->get_features_test());
 	auto test_labels = wrap(env->get_labels_test());
 
-	auto perceptron = some<CPerceptron>(features, labels);
-	perceptron->train();
+	auto perceptron = some<CPerceptron>();
+	perceptron->train(features, labels);
 
 	auto weights = perceptron->get_w();
 
-	auto perceptron_initialized = some<CPerceptron>(features, labels);
+	auto perceptron_initialized = some<CPerceptron>();
 	perceptron_initialized->set_initialize_hyperplane(false);
 	perceptron_initialized->set_w(weights);
 	perceptron_initialized->set_max_iter(1);
 
-	perceptron_initialized->train();
+	perceptron_initialized->train(features, labels);
```

Comment:
pls use the base class method here (this means you need to manually set labels before training)